### PR TITLE
chore: remove unused code

### PR DESCRIPTION
**Pull request by incendial**
This PR removes some of the unused code reported here: https://github.com/flutter/devtools/issues/5139. There will probably be another PR when I fix some false-positives with conditional imports.

**Note:** not all of the removed code might be unused intentionally, but I have no context of that. Also, there are might be  not obvious to me false-positives.

RELEASE_NOTE_EXCEPTION=[Internal refactoring] 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [-] I updated/added relevant documentation (doc comments with `///`).
- [-] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

Not sure, what's expected of me to do with the badge above, could you please help to understand that?

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart
Comment (jacob314) [inline packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart:10]: Revert changes to this file. It is generated.

**Comment by incendial** [comment]
@jacob314 I've updated the PR after the analyzer reports, could you please restart the pipelines?

File: packages/devtools_app/lib/src/screens/debugger/codeview.dart
Comment (incendial) [inline packages/devtools_app/lib/src/screens/debugger/codeview.dart:551]: Removal was done automatically, so some newlines might be absent. Will restore this one while applying other review comments.

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:174]: here and below: @polina-c this looks like a bug that we are not passing this through to the context menu button

File: packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart:10]: Ah, I see, now it fails coz of that. Will update tomorrow.

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:174]: The context is supposed to be consumed by InstanceSetButton, that is under construction. @incendial , is there a way to exempt  gaContext from clean up?

Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:174]: There are several ways to tell the command to ignore some code:
- via `--exclude` config to ignore a whole file
- via `ignore_for_file: unused_code`
- via `ignore: unused_code`

but I think I can just revert this removal manually and this should be enough.

File: packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/provider/instance_viewer/fake_freezed_annotation.dart]: I've reverted this change, could you please restart the pipelines?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:174]: But my idea is to make the tests green, then you'll probably point out some other places that should be reverted and I want to apply all the review comments at once. Does it sound good?

Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart:174]: Sure. Yes, you can revert the change. Or, you can add `ignore: unused_code` for  gaContext in InstanceSetButton.

**Comment by incendial** [comment]
So looks like everything is green. I'm waiting on the comments on what should be reverted then.

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_single.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_single.dart:112]: revert this change, too

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/instances.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/instances.dart:7]: revert changes to this file

File: packages/devtools_app/lib/src/screens/memory/panes/leaks/diagnostics/leak_analyzer.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/leaks/diagnostics/leak_analyzer.dart:48]: @polina-c should this be reverted?

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_button.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_button.dart:9]: revert changes to this file

File: packages/devtools_app/lib/src/screens/network/network_request_inspector.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/network/network_request_inspector.dart:30]: fine to remove this, but can you add a TODO to clean to remove these keys and find widgets by text  in our tests? you can assign to me

`// TODO(kenz): remove these keys and use a text finder to lookup widgets in test.`

File: packages/devtools_app/lib/src/screens/performance/panes/rebuild_stats/rebuild_stats_model.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/rebuild_stats/rebuild_stats_model.dart:71]: @jacob314 should this be removed? or is this just not used yet since this feature is under construction?

File: packages/devtools_app/lib/src/screens/provider/instance_viewer/result.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/provider/instance_viewer/result.dart:26]: please remove these changes since this file is generated

File: packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart:762]: generally curious since the `_analytics_web.dart` file is imported through conditional imports where it is used in DevTools - does the check_unused_code command work well with conditional imports?

File: packages/devtools_app/lib/src/shared/analytics/constants.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/analytics/constants.dart:79]: this is a bug that this is unused. Could you please replace `shaderCompilationDocs` with `intrinsicOperationsDocs` here instead of removing? https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/performance/panes/frame_analysis/frame_hints.dart#L269

File: packages/devtools_app/lib/src/shared/analytics/constants.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/analytics/constants.dart:212]: @polina-c is it a bug that these are unused?

File: packages/devtools_app/lib/src/shared/charts/chart_trace.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/charts/chart_trace.dart:184]: FYI @polina-c this now appears to be a strange pattern we are using a ValueNotifier for. Seems like maybe we should be using a stream instead? This is used to notify the memory chart that we need to recompute. We should file a bug to clean up.

File: packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart:64]: can we revert this change and the same for the _server_web.dart file below? This is the DevTools API to access the DevTools server, so even if it is not used right now, leaving this in the interface is still useful to expose the DevTools server endpoints. (kind of like the vm_service_wrapper.dart file - we don't use every method, but we are implementing a protocol so we leave them all there)

File: packages/devtools_app/lib/src/shared/framework_controller.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/framework_controller.dart:39]: please revert these changes. ConnectVmEvent's fields are actually used here, but it looks like these are getting flagged as unused because we are mistakenly using dynamic execution:
https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/framework/initializer.dart#L107-L112

Can you please add the type `ConnectVmEvent` to the parameter event in this method? and then revert these changes. thanks

File: packages/devtools_app/lib/src/shared/http/_http_date.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/http/_http_date.dart:106]: @CoderDake @bkonyi is this a bug that this is unused?

File: packages/devtools_app/lib/src/shared/server_api_client.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/server_api_client.dart:157]: revert (see earlier comment about the ConnectVmEvent class)

File: packages/devtools_app/test/legacy_integration_tests/debugger.dart
Comment (kenzieschmoll) [inline packages/devtools_app/test/legacy_integration_tests/debugger.dart:403]: can you ignore changes in legacy_integration_tests/ for now? These tests need to be re-written as real integration tests, and the existing code will be useful for porting these over, even if it is not used for now (some of these tests are disabled or commented out at the moment)

File: packages/devtools_app/test/shared/widget_rebuild_counts_test.dart
Comment (kenzieschmoll) [inline packages/devtools_app/test/shared/widget_rebuild_counts_test.dart:122]: please leave this for now, as this feature is under construction and this locationJson helper may be used when completing the test coverage for this feature

File: packages/devtools_app/test/test_infra/scenes/performance/default.dart
Comment (kenzieschmoll) [inline packages/devtools_app/test/test_infra/scenes/performance/default.dart:50]: please leave this. This is a bug that we are not calling tearDown on the scene from tests that use this stager scene

**Review by kenzieschmoll — COMMENTED** [review]
Looking good! I reviewed and left comments throughout. Looks like you'll need to merge to resolve some conflicts.

**Comment by incendial** [comment]
@kenzieschmoll hey, thank you for the review! Should we wait until others reply / review? I'd like to avoid multiple iterations and merges if possible.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/screens/performance/panes/rebuild_stats/rebuild_stats_model.dart
Comment (jacob314) [reply packages/devtools_app/lib/src/screens/performance/panes/rebuild_stats/rebuild_stats_model.dart:71]: should be removed

File: packages/devtools_app/lib/src/shared/charts/chart_trace.dart
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/charts/chart_trace.dart:186]: This is very old code that needs cleanup. If/when somebody needs to change it, they will do it. Not sure bug will make change. Added TODO.

File: packages/devtools_app/lib/src/screens/memory/shared/heap/heap.dart
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/shared/heap/heap.dart:141]: If the change compiles, it is ok to delete.

File: packages/devtools_app/lib/src/shared/analytics/constants.dart
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/analytics/constants.dart:212]: See https://github.com/flutter/devtools/pull/5301/files#r1117500947

File: packages/devtools_app/lib/src/shared/http/_http_date.dart
Comment (CoderDake) [reply packages/devtools_app/lib/src/shared/http/_http_date.dart:106]: I can't seem to find any uses of it anywhere.
Something about it seems like it was copied from somewhere anyways, especially if we don't have tests for it. As such it would probably be brittle enough that it would be ok to let go of.

File: packages/devtools_app/test/test_infra/flutter_test_driver.dart
Comment (CoderDake) [inline packages/devtools_app/test/test_infra/flutter_test_driver.dart:51]: These could be helpful for debugging in the future. We could probably leave them.

File: packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart:774]: I've fixed almost all false-positives with conditional imports (the only one left is when one conditionally import uses publicly available code from its file and another (main) does not). Unfortunately, analyzer resolves conditional imports not the way I'd expected it to be, it's calculated manually on top.

But as for this change, this code is unused. Or am I missing something?

File: packages/devtools_app/test/test_infra/scenes/performance/default.dart
Comment (incendial) [reply packages/devtools_app/test/test_infra/scenes/performance/default.dart:50]: Should I add TODO?

File: packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/classes_table_diff.dart]: I've reverted all changes with gaContext

File: packages/devtools_app/lib/src/screens/provider/instance_viewer/result.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/provider/instance_viewer/result.dart]: Done

File: packages/devtools_app/lib/src/shared/feature_flags.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/feature_flags.dart]: Done

File: packages/devtools_app/lib/src/shared/framework_controller.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/framework_controller.dart]: Wasn't expecting that 😅. Done

File: packages/devtools_app/test/shared/widget_rebuild_counts_test.dart
Comment (incendial) [reply packages/devtools_app/test/shared/widget_rebuild_counts_test.dart]: Done

File: packages/devtools_app/lib/src/screens/inspector/layout_explorer/ui/widgets_theme.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/inspector/layout_explorer/ui/widgets_theme.dart]: Reverted

**Comment by polina-c** [comment]
> 

Thanks. Will remove them in separate PR.

**Review by polina-c — APPROVED** [review]

**Comment by polina-c** [comment]
You want to pull latest changes to to get fix for the failing golden test.

File: packages/devtools_app/test/test_infra/flutter_test_driver.dart
Comment (incendial) [reply packages/devtools_app/test/test_infra/flutter_test_driver.dart]: Thanks, done!

**Comment by incendial** [comment]
> You want to pull latest changes to to get fix for the failing golden test.

Thank you, done, hope they'll pass

File: packages/devtools_app/test/legacy_integration_tests/debugger.dart
Comment (kenzieschmoll) [reply packages/devtools_app/test/legacy_integration_tests/debugger.dart]: no worries - we already have a TODO at the top of the intetgration_test.dart file to rewrite the integration tests: https://github.com/flutter/devtools/issues/1987

File: packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/analytics/_analytics_web.dart:774]: This code appears to be unused 👍

File: packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart]: Yes, please add a comment. Thanks.

**Review by kenzieschmoll — APPROVED** [review]

**Comment by kenzieschmoll** [comment]
If you merge master are you still seeing the golden failures? The golden image tests are passing at head on a different PR from this afternoon.

File: packages/devtools_app/test/test_infra/scenes/performance/default.dart
Comment (incendial) [reply packages/devtools_app/test/test_infra/scenes/performance/default.dart]: Done

File: packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/config_specific/server/_server_stub.dart]: Done

**Comment by incendial** [comment]
> If you merge master are you still seeing the golden failures?

With Dart 2.19.4 / Flutter 3.7.7 it's not possible to resolve the dependencies, so I'm only relying on the CI to run the test. Can someone restart the workflow?
> Because every version of integration_test from sdk depends on vm_service 9.4.0 and devtools_app depends on vm_service ^11.2.0, integration_test from sdk is forbidden.

**Comment by incendial** [comment]
Looks like all the tests passed. Is there anything else to be done to unblock the merge?

**Comment by CoderDake** [comment]
@incendial Thank you so much for all of your hard work on this 😄 🎉 🤠 🥳 🎊

**Comment by incendial** [comment]
Thank you! 🔥🚀

**Pull request merged**
